app/src/main/java/com/example/myapplication/utils/OnScrollListener.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.myapplication.utils
+
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+
+class OnScrollListener(val layoutManager: LinearLayoutManager, val getMoreData: () -> Unit) : RecyclerView.OnScrollListener() {
+    var previousTotal = 0
+    var loading = true
+    val visibleThreshold = 10
+    var firstVisibleItem = 0
+    var visibleItemCount = 0
+    var totalItemCount = 0
+
+    override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+        super.onScrolled(recyclerView, dx, dy)
+
+        visibleItemCount = recyclerView.childCount
+        totalItemCount = recyclerView.layoutManager!!.itemCount
+        firstVisibleItem = (recyclerView.layoutManager as LinearLayoutManager).findFirstVisibleItemPosition()
```

Comment:
You should get used to kotlin functions like `.also`, `.let`, `.run`, `.apply` and `with`... Which one could you use here?